examples/windows_nodepool_with_private_cluster/modules/google_kubernetes_engine/gke_node_pool/main.tf

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This file was automatically generated from a template in ./autogen/main
+
+locals {
```

Comment:
These should not be added. We should be using one our existing `private-cluster` modules.

Reply:
This is custom module so support windows nodepool on GKE cluster, hance added modules section here in repo only.
let me know if that works

Reply:
Hi @HarshalRane23
We should leverage the existing module and use it in an example with windows nodepools.  An example config can be
```hcl
module "gke" {
  source                            = "../../modules/beta-public-cluster/"
  project_id                        = var.project_id
  name                              = "widows-pool-${var.cluster_name_suffix}"
region                    = var.region
  network                 = var.network
  subnetwork              = var.subnetwork
  ip_range_pods           = var.ip_range_pods
  ip_range_services       = var.ip_range_services
  service_account         = var.compute_engine_service_account
  enable_private_endpoint = true
  enable_private_nodes    = true
  network_policy = false
  master_ipv4_cidr_block  = "172.16.0.0/28"
  enable_shielded_nodes = false
  node_metadata = "EXPOSE"
  identity_namespace = null
  

  master_authorized_networks = [
    {
      cidr_block   = data.google_compute_subnetwork.subnetwork.ip_cidr_range
      display_name = "VPC"
    },
  ]
    node_pools = [
    {
      name               = "linux-node-pool"
      min_count          = 1
      max_count          = 10
      disk_size_gb       = 100
      disk_type          = "pd-standard"
      image_type         = "COS"
      initial_node_count = 1
    },
    {
      name               = "windows-node-pool"
      min_count          = 1
      max_count          = 10
      disk_size_gb       = 100
      disk_type          = "pd-standard"
      image_type         = "WINDOWS_SAC"
      initial_node_count = 1
      enable_integrity_monitoring = false
    }
  ]
}
```